src/sagemaker/workflow/model_step.py

```diff
@@ -0,0 +1,217 @@
+# may not use this file except in compliance with the License. A copy of
+# the License is located at
+#
+#     http://aws.amazon.com/apache2.0/
+#
+# or in the "license" file accompanying this file. This file is
+# distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF
+# ANY KIND, either express or implied. See the License for the specific
+# language governing permissions and limitations under the License.
+"""The `ModelStep` definition for SageMaker Pipelines Workflows"""
+from __future__ import absolute_import
+
+import logging
+from typing import Union, List, Dict, Optional
+
+from sagemaker import Model, PipelineModel
+from sagemaker.model import SCRIPT_PARAM_NAME, DIR_PARAM_NAME
+from sagemaker.workflow import Properties
+from sagemaker.workflow._utils import _RegisterModelStep, _RepackModelStep
+from sagemaker.workflow.pipeline_context import PipelineSession, _ModelStepArguments
+from sagemaker.workflow.retry import RetryPolicy
+from sagemaker.workflow.step_collections import StepCollection
+from sagemaker.workflow.steps import Step, CreateModelStep
+
+NEED_RUNTIME_REPACK = "need_runtime_repack"
+
+_CREATE_MODEL_RETRY_POLICIES = "create_model_retry_policies"
+_REGIS_MODEL_RETRY_POLICIES = "register_model_retry_policies"
+_REPACK_MODEL_RETRY_POLICIES = "repack_model_retry_policies"
+
+
+class ModelStep(StepCollection):
+    """`ModelStep` for SageMaker Pipelines Workflows."""
+
+    def __init__(
+        self,
+        name: str,
+        step_args: dict,
+        depends_on: Optional[Union[List[str], List[Step]]] = None,
+        retry_policies: Optional[Union[List[RetryPolicy], Dict[str, List[RetryPolicy]]]] = None,
+        display_name: Optional[str] = None,
+        description: Optional[str] = None,
+    ):
+        """Constructs a `ModelStep`.
+
+        Args:
+            name (str): The name of the `ModelStep`. A name is required and must be
+                unique within a pipeline.
+            step_args (dict): The arguments for the `ModelStep` definition.
+            depends_on (List[str] or List[Step]): A list of `Step` names or `Step` instances
+                that this `ModelStep` depends on.
+                If a listed `Step` name does not exist, an error is returned (default: None).
+            retry_policies (List[RetryPolicy] or Dict[str, List[RetryPolicy]]): The list of retry
+                policies for the `ModelStep`.
+            display_name (str): The display name of the `ModelStep`.
+                The display name provides better UI readability. (default: None).
+            description (str): The description of the `ModelStep` (default: None).
+        """
+        if not isinstance(step_args, _ModelStepArguments):
+            raise TypeError(
+                "To correctly configure a ModelStep, "
+                "the step_args must be a `_ModelStepArguments` object generated by "
+                "`.create()` or `.register()`."
```

Comment:
nit: will docstring supports markdown? namely, translate the "`.create()`"

Reply:
Actually it's not markdown and it's just add italic format to the word. I learnt this while having my FailStep PR under review.
The doc string will be auto populated to [here](https://sagemaker.readthedocs.io/en/stable/workflows/pipelines/sagemaker.workflow.pipelines.html#sagemaker.workflow.fail_step.FailStep)
As you can see, as `FailStep` has the italic format.

Reply:
Okay, just realize this is an error message rather than a doc string. Will remove the ` in next change once I collect more feedback.